app.polychromatic.controller.json

```diff
@@ -0,0 +1,184 @@
+{
+    "app-id": "app.polychromatic.controller",
+    "runtime": "org.kde.Platform",
+    "runtime-version": "6.7",
+    "sdk": "org.kde.Sdk",
+    "command": "polychromatic-controller",
+    "base": "com.riverbankcomputing.PyQt.BaseApp",
+    "base-version": "6.7",
+    "rename-icon": "polychromatic",
+    "rename-desktop-file": "polychromatic.desktop",
+    "cleanup-commands": [
+        "/app/cleanup-BaseApp.sh"
+    ],
+    "finish-args": [
+        "--env=QTWEBENGINEPROCESS_PATH=/app/bin/QtWebEngineProcess",
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--share=ipc",
+        "--share=network",
+        "--talk-name=org.razer",
```

Comment:
org.razer -> what is this for?

Reply:
that's the dbus service name of the openrazer daemon https://openrazer.github.io/ which does the actual communication to USB. OpenRazer needs to be installed via the distro since it also contains kernel modules and more.

Reply:
Yep ^^ Polychromatic is just the frontend, it uses the OpenRazer Python library, which communicates over D-Bus to a process `openrazer-daemon` running on the host — which in turn would echo to the driver using the sysfs filesystem (amongst other things).

This requirement is mentioned in this app's [metainfo description.](https://github.com/polychromatic/polychromatic/blob/8861b76986858690c73d525258bfd5a8511dd3ad/sources/app.polychromatic.controller.metainfo.xml#L43-L46) 

If you don't have Razer hardware to test, if OpenRazer is installed on the host, you can use `scripts/setup_fake_devices.sh` from the [repository](https://github.com/openrazer/openrazer).